aptos-move/framework/move-stdlib/src/natives/signer.rs

```diff
@@ -34,6 +37,16 @@ fn native_borrow_address(
 
     let signer_reference = safely_pop_arg!(arguments, SignerRef);
 
+    if context
```

Comment:
I think this should just dynamically check whether the caller is from 0x1, adding a new `context.interpreter.called_from(&AccountAddress)` or similar, and this PR would become a lot simpler, not changing any Move code, which we may need to change back when using better APIs then `permissioned_signer::address_of`.

Reply:
+1 this is a great idea and can potentially lessen the devex impact here